projects/storefrontlib/cms-components/myaccount/my-account-v2/index.ts

```diff
@@ -6,3 +6,6 @@
 
 export * from './my-account-v2-navigation/index';
 export * from './my-account-v2.module';
+export * from './my-account-v2-consent-management/index';
+export * from '../my-account-v2-notification-preference/index';
```

Comment:
Hi @Platonn
consent-management.component - v2 and v1 has HTML DOM structure , v2 added some buttons and changed almost all the styles
notification-preference.component - v2 and v1 has HTML DOM structure, v2 added a header

Reply:
Thank you for the explanation. I missed the fact that that HTML/CSS are different. I can however that TS are identical.

I'm still concerned about adding new kilobytes of code to the `main.js` bundle, that will be loaded for every user on the first page's visit.

As far as I understand, the components of Consent Management and Notification Preferences are displayed actually only for logged-in users and only when they enter the `/my-account/...` route. If yes, then fundamentally they don't belong to the `@spartacus/storefront` library but more to `@spartacus/user` library - which is lazy loadable.

Up until now we didn't split those components to the user library.
But since we're copy-pasting those components anyway, what do you think about placing those v2 components to the `@spartacus/user` library? (e.g. `@spartacus/user/profile or `@spartacus/user/account` - I'm not sure which one is more appropriate). Of course, then the CMS sample data should be different: we won't map the Angular implementations of v2 components anymore to the old CMS component types. But for v2, a brand new, different strings CMS component types should be used. And the old cms component types should be removed from sample data in such a case - but only if our customer wants to use v2.

On one hand it would be great to place new components in the more appropriate place - in a lazy laodable library instead of the storefront lib.
On the other hand, complicates the setup of v2 versus v1. It would require that on routes /notification-preferences and /consent-management the CMS sample data (the CMS component types) should be served differently for a storefront that wants to use MyAccount v1 versus a storefront that wants to use MyAccount v2.

Because of the requirement to change conditionally sample data, I perceive this whole improvement cumbersome and not a low-hanging fruit.

Therefore let me propose a different solution for minimizing the footprint of additional KB in the `main.js` bundle:

### Inheriting TS of components v1 in components v2
I can see that TS of components v2 are identical as TS of components v1. This copied source code contributes to the additional footprint of KB in the `main.js` bundle, which is probably not necessary. I have an idea that we could simply extend the v1 class by the v2 class, and only define a custom HTML. For example:

```ts
@Component({
  selector: 'cx-my-account-v2-consent-management',
  templateUrl: './my-account-v2-consent-management.component.html',
})
export class MyAccountV2ConsentManagementComponent extends ConsentManagementComponent {}
```

Could you please try if it works for you?

PS. in case of `consentApprovedTime` (inside `MyAccountV2ConsentManagementFormComponent`) we can just add this logic seamlessly to the TS of component v1, so it's inherited in v2. IMHO it's a fair tradeoff - it's less KB footprint than duplicating the whole TS of class v1 to the class v2.

Note: this applies to all components in the `@spartacus/storefront` library, because they all have (almost) identical TS of components v1 to components v2